lib/vquic/msh3.c

```diff
@@ -512,14 +622,43 @@ static CURLcode cf_msh3_connect(struct Curl_cfilter *cf,
   return result;
 }
 
-static void cf_msh3_ctx_clear(struct cf_msh3_ctx *ctx)
+static void cf_msh3_ctx_clear(struct Curl_cfilter *cf, struct Curl_easy *data)
 {
+  struct cf_msh3_ctx *ctx = cf->ctx;
+
+  (void)data;
+  CF_DEBUGF(infof(data, CFMSG(cf, "clearing ctx")));
   if(ctx) {
     if(ctx->qconn)
       MsH3ConnectionClose(ctx->qconn);
     if(ctx->api)
       MsH3ApiClose(ctx->api);
+
+    if(ctx->active) {
+      /* We share our socket at cf->conn->sock[cf->sockindex] when active.
+       * If it is no longer there, someone has stolen (and hopefully
```

Comment:
"someone has stolen [it]"?? Could you explain this some?

Reply:
Ideally, the connection filters that create sockets should also be the ones to close them. However, there are code paths on curl that access the `conn->sock[index]` ones and seem to do that. I need to track those cases down.

Reply:
That doesn't sound good. If we do expose the socket from MsQuic up to curl and it closes it, I expect really bad things would happen.